e2e/cypress/integration/collapsed_reply_threads/replies_spec.js

```diff
@@ -0,0 +1,127 @@
+            testTeam = team;
+            testUser = user;
+            testChannel = channel;
+
+            cy.apiSaveCRTPreference(testUser.id, 'on');
+            cy.apiCreateUser({prefix: 'other'}).then(({user: user1}) => {
+                otherUser = user1;
+
+                cy.apiAddUserToTeam(testTeam.id, otherUser.id).then(() => {
+                    cy.apiAddUserToChannel(testChannel.id, otherUser.id);
+
+                    // # Post a message as other user
+                    cy.postMessageAs({sender: otherUser, message: 'Root post', channelId: testChannel.id}).then((post) => {
+                        rootPost = post;
+                    });
+                });
+            });
+        });
+    });
+
+    beforeEach(() => {
+        // # Visit the channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+    });
+
+    it('should show number of replies in thread', () => {
+        // # Thread footer should not exist
+        cy.get(`#post_${rootPost.id}`).find('.ThreadFooter').should('not.exist');
+
+        // # Post a root post as current user
+        cy.postMessageAs({sender: testUser, message: 'reply!', channelId: testChannel.id, rootId: rootPost.id});
+
+        // # Get last root post
+        cy.get(`#post_${rootPost.id}`).
+
+            // * Get Thread footer
+            get('.ThreadFooter').
+            within(() => {
+                // * Reply button in Thread Footer should say '1 reply'
+                cy.get('.ReplyButton').should('have.text', '1 reply');
+
+                // * 2 avatars/participants should show in Thread Footer
+                cy.get('.Avatar').should('have.lengthOf', 2);
+            });
+
+        // # Visit global threads
+        cy.visit(`/${testTeam.name}/threads`);
+
+        // * The sole thread item should have text in footer saying '1 reply'
+        cy.get('article.ThreadItem').find('.activity').should('have.text', '1 reply');
+
+        // # Visit the channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+        // # Post another reply as current user
+        cy.postMessageAs({sender: testUser, message: 'another reply!', channelId: testChannel.id, rootId: rootPost.id});
+
+        // # Get last root post
+        cy.get(`#post_${rootPost.id}`).
+
+            // # Get Thread footer
+            get('.ThreadFooter').
+            within(() => {
+                // * Reply button in thread footer should say '2 replies'
+                cy.get('.ReplyButton').should('have.text', '2 replies');
+
+                // * 2 avatars/participants should show in Thread Footer
+                cy.get('.Avatar').should('have.lengthOf', 2);
+            });
+
+        // # Visit global threads
+        cy.visit(`/${testTeam.name}/threads`);
+
+        // * The sole thread item should have text in footer saying '2 replies'
+        cy.get('article.ThreadItem').find('.activity').should('have.text', '2 replies');
```

Comment:
yep we should, but lets add it separately since we are going to change the avatars in global threads to not show the participant in https://mattermost.atlassian.net/browse/MM-35017 and we should then also check the ordering

Reply:
Thanks Kyriakos I'll add a note to the ticket.